internal/controllers/user_devices_controller.go

```diff
@@ -1445,15 +1440,45 @@ func (udc *UserDevicesController) PostMintDevice(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusBadRequest, "Empty image field.")
 	}
 
-	requestID := ksuid.New().String()
+	var hash []byte
+	if udc.Settings.IsProduction() {
+		mvs := registry.MintVehicleSign{
+			ManufacturerNode: makeTokenID,
+			Owner:            common.HexToAddress(*user.EthereumAddress),
+			Attributes:       []string{"Make", "Model", "Year"},
+			Infos:            []string{dd.Make.Name, dd.Type.Model, strconv.Itoa(int(dd.Type.Year))},
+		}
 
-	logger.Info().
-		Interface("httpRequestBody", mr).
-		Interface("client", client).
-		Interface("mintVehicleSign", mvs).
-		Interface("typedData", client.GetPayload(&mvs)).
-		Msg("Got request.")
+		logger.Info().
+			Interface("httpRequestBody", mr).
+			Interface("client", client).Interface("mintVehicleSign", mvs).
+			Interface("typedData", client.GetPayload(&mvs)).
+			Msg("Got request.")
 
+		hash, err = client.Hash(&mvs)
+		if err != nil {
+			return opaqueInternalError
+		}
+	} else {
+		mvs := registry.MintVehicleWithDeviceDefinitionSign{
+			ManufacturerNode:   makeTokenID,
+			Owner:              common.HexToAddress(*user.EthereumAddress),
+			DeviceDefinitionID: userDevice.DeviceDefinitionID,
```

Comment:
A realization: this is not the device definition the chain expects. We'll have to convert.

Reply:
oh yeah, great catch!